src/org/zaproxy/zap/extension/authStats/ExtensionAuthStats.java

```diff
@@ -0,0 +1,159 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.zaproxy.zap.extension.authStats;
+
+import java.util.regex.Pattern;
+
+import org.apache.commons.httpclient.URI;
+import org.apache.commons.httpclient.URIException;
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.extension.ExtensionAdaptor;
+import org.parosproxy.paros.extension.ExtensionHook;
+import org.parosproxy.paros.model.Model;
+import org.parosproxy.paros.model.Session;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpSender;
+import org.parosproxy.paros.network.HttpStatusCode;
+import org.zaproxy.zap.authentication.AuthenticationMethod;
+import org.zaproxy.zap.model.Context;
+import org.zaproxy.zap.model.SessionStructure;
+import org.zaproxy.zap.network.HttpSenderListener;
+import org.zaproxy.zap.utils.Stats;
+
+/**
+ * Authentication Statistics - Records logged in/out statistics for all contexts in scope 
+ * @author psiinon
+ *
+ */
+public class ExtensionAuthStats extends ExtensionAdaptor implements HttpSenderListener {
+    
+    private static final Logger log = Logger.getLogger(ExtensionAuthStats.class);
+
+    @Override
+    public void hook(ExtensionHook extensionHook) {
+        super.hook(extensionHook);
+        HttpSender.addListener(this);
+    }
+
+    @Override
+    public String getAuthor() {
+        return Constant.ZAP_TEAM;
+    }
+
+    @Override
+    public String getName() {
+        return "ExtensionAuthStats";
+    }
+
+    @Override
+    public String getUIName() {
+        return Constant.messages.getString("authStats.name");
+    }
+
+    @Override
+    public String getDescription() {
+        return Constant.messages.getString("authStats.desc");
+    }
+    
+    @Override
+    public boolean canUnload() {
+        // TODO change when unload() can be implemented
+        return false;
+    }
+
+    @Override
+    public void unload() {
+        super.unload();
+        // TODO change to use HttpSender.removeListener when available
+    }
+
+    @Override
+    public int getListenerOrder() {
+        return 0;
+    }
+
+    @Override
+    public void onHttpRequestSend(HttpMessage msg, int initiator, HttpSender sender) {
+        // Ignore        
+    }
+
+    @Override
+    public void onHttpResponseReceive(HttpMessage msg, int initiator, HttpSender sender) {
+        String comp;
+        switch (initiator) {
```

Comment:
Should ignore requests from `AUTHENTICATION_INITIATOR`? If not, it should probably be replaced with "auth" component?